app/uk/gov/hmrc/customs/declaration/services/XmlValidationService.scala

```diff
@@ -48,45 +47,26 @@ abstract class XmlValidationService @Inject()(val configuration: Configuration,
 
   private lazy val maxSAXErrors = configuration.getOptional[Int]("xml.max-errors").getOrElse(Int.MaxValue)
 
+  private lazy val validator = new ValidateXmlAgainstSchema(schema)
```

Comment:
Could DI be used rather than `new`?

Reply:
This instance of ValidateXmlAgainstSchema is unique to this instance of XmlValidationService as the 'schema' is memoized by the new validator.